src/main/java/org/pitest/junit5/ConditionalTestFilter.java

```diff
@@ -0,0 +1,48 @@
+package org.pitest.junit5;
+
+import org.junit.jupiter.api.extension.ConditionEvaluationResult;
+import org.junit.jupiter.api.extension.ExecutionCondition;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+/**
+ * A test filter accepting all tests until requested to reject
+ * all subsequent tests through an {@linkplain TestUnitExecutionsRegistry executions registry}.
+ */
+public class ConditionalTestFilter implements ExecutionCondition {
+
+    static final String EXECUTION_ID_KEY = "org.pitest.junit5.executionId";
+
+    private static final ConditionEvaluationResult DISABLED = ConditionEvaluationResult.disabled(
+            "Pitest has requested to skip remaining tests in a test unit");
+    private static final ConditionEvaluationResult ENABLED = ConditionEvaluationResult.enabled(null);
+
+    private final TestUnitExecutionsRegistry executionsRegistry;
+
+    @SuppressWarnings("unused") // Used through reflection by the ServiceLoader
+    // when this extension is requested by JUnit Jupiter Engine
+    public ConditionalTestFilter() {
+        this(TestUnitExecutionsRegistry.getInstance());
```

Comment:
I am not entirely happy with this global registry, but haven't found another way to inject an object into the extension (each created launcher instantiates an engine which, in turn, creates a new extension — one cannot supply an extension through launcher, because it is not aware of engine-specific extensions).

Reply:
I don't have a better solution, but I'd be very reluctant to merge in anything that introduced global state.

Just had a scan of the junit 5 apis and the "smaller test unit" approach might be easier for junit 5 - assuming that parameterized tests etc are identifiable with their own ids during the scan stage.

Reply:
The only way to know if the performance impact is significant would be to run some experiments, but as I think you are probably correct that the individual ids will not be discoverable at the scan stage the point is moot.

Which takes us back to trying to break out of the running test again.

Reply:
Some benchmarks of the various options are essential indeed.

There is a related discussion regarding APIs to terminate dynamic (a subset of test containers) tests early, where I tried to summarize the issue with the approach in the present prototype: https://github.com/junit-team/junit5/issues/431